src/app/pages/payment/pin-payment.component.ts

```diff
@@ -0,0 +1,102 @@
+import { validateInput } from 'src/app/validators/payment-validators';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, Validators, FormControl } from '@angular/forms';
+import { PaymentService } from 'src/app/services/payment-service';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { Location } from '@angular/common';
+import { Title } from '@angular/platform-browser';
+
+@Component({
+  selector: 'app-pin-payment',
+  templateUrl: './pin-payment.component.html',
+  styleUrls: ['./payment.component.scss']
+})
+export class PinPaymentComponent implements OnInit {
+  validYears: number[];
+  validMonths: string[] = [
+    '01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'
+  ];
+  cardForm = new FormGroup({
+    cardNumber: new FormControl('', Validators.required),
+    securityNumber: new FormControl('', Validators.required),
+    cardExpiry: new FormGroup({
+      expiryMonth: new FormControl(
+        'MM', [Validators.required, validateInput('MM')]
+        ),
+      expiryYear: new FormControl(
+        'YYYY', [Validators.required, validateInput('YYYY')]
+        )
+    }),
+    amount: new FormControl('', [Validators.required, Validators.min(0)]),
+    pin: new FormControl('', Validators.required),
+    purpose: new FormControl('Purpose', [Validators.required, validateInput('Purpose')]),
+    saveCard: new FormControl(false)
+  });
+
+  constructor(
+    private paymentService: PaymentService,
+    private router: Router,
+    private spinner: NgxSpinnerService,
+    private toastr: ToastrService,
+    private location: Location,
+    private titleService: Title
+    ) {this.titleService.setTitle('Card & Pin Deposit'); }
+
+  ngOnInit(): void {
+    this.generateValidYears();
+  }
+
+  generateValidYears() {
+    const yearList: number[] = [];
+    const now = new Date();
+    const currentYear = now.getFullYear();
+    const lastValidYear = currentYear + 10;
+    for (let i = currentYear; i <= lastValidYear; i++ ) {
+      yearList.push(i);
+    }
+    this.validYears = yearList;
+  }
+
+  submitDetails() {
+    const formValue = this.cardForm.value;
+    const payload = {
+      cardno: formValue.cardNumber,
+      cvv: formValue.securityNumber,
+      expirymonth: formValue.cardExpiry.expiryMonth,
+      expiryyear: formValue.cardExpiry.expiryYear.slice(-2),
+      amount: formValue.amount,
+      pin: formValue.pin,
+      purpose: formValue.purpose,
+      save_card: formValue.saveCard
+    };
+    this.spinner.show();
+    this.paymentService.initiatePinPay(payload).subscribe(
+      resp => {
+        this.spinner.hide();
+        this.router.navigate(['/validate-pin', resp['flwRef'],
+        formValue.purpose]);
+      },
+      error => {
```

Comment:
Great job  @Victoradukwu working on this functionality I like the fact that you included a dummy card number so that I didn't have to search my wallet for my card 😆 , however, the form has a small issue if a user enters more than 5 digits in the security number an error is returned from the API but the user isn't notified of the same for instance in my case it should tell me the field should not have more than 5 digits but it just shows the loading indicator and the user gets no feedback.

![image](https://user-images.githubusercontent.com/12128153/63412039-8b244e80-c3ff-11e9-83fe-3a911cc38b72.png)

I believe this is brought about because of the shape of the error object being returned by the `cvv` invalidation in the backend. Check the screenshot below of the error being returned for more clariry

![image](https://user-images.githubusercontent.com/12128153/63412555-a479ca80-c400-11e9-8bc2-73ef9dc6380b.png)

Reply:
Thanks, @Tevinthuku  for spotting that creepy serializer error. It is fixed.